recipes/onnxruntime/all/conanfile.py

```diff
@@ -0,0 +1,348 @@
+            value = "ON" if value else "OFF"
+
+        self._cmake.definitions[f"onnxruntime_{definition}"] = value
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._add_definition("PREFER_SYSTEM_LIB", True)
+
+        with_tests = self.options.with_tests
+        if not tools.cross_building(self):
+            self._add_definition("RUN_ONNX_TESTS", with_tests)
+            self._add_definition("RUN_MODELTEST_IN_DEBUG_MODE", with_tests)
+            self._add_definition("FUZZ_TEST", with_tests)
+
+        self._add_definition("USE_VALGRIND", False)
+
+        self._add_definition("BUILD_SHARED_LIB", self.options.shared)
+        self._add_definition("BUILD_UNIT_TESTS", self.options.with_tests)
+        self._add_definition("BUILD_BENCHMARKS", self.options.with_benchmarks)
+        self._add_definition("BUILD_FOR_NATIVE_MACHINE", False)  # ??
+        self._add_definition("BUILD_WEBASSEMBLY", False)
+
+        self._add_definition("GENERATE_TEST_REPORTS", self.options.with_tests)
+
+        self._add_definition("ENABLE_STATIC_ANALYSIS", False)
+        self._add_definition("ENABLE_MEMLEAK_CHECKER", False)
+        self._add_definition("ENABLE_CUDA_LINE_NUMBER_INFO", False)  # True may cause LNK1169 error
+        self._add_definition("ENABLE_LTO", False)
+        self._add_definition("ENABLE_INSTRUMENT", False)
+        self._add_definition("ENABLE_WEBASSEMBLY_EXCEPTION_CATCHING", False)
+
+        self._add_definition("ENABLE_PYTHON", self.options.with_python)
+        self._add_definition("BUILD_CSHARP", self.options.with_csharp)
+        self._add_definition("USE_CUDA", self.options.with_cuda)
+
+        self._add_definition("USE_OPENVINO", self.options.with_openvino)
+
+        if self.settings.os == "iOS" or self.settings.os == "Macos":
+            self._add_definition("USE_COREML", self.options.with_coreml)
+        if self.settings.os == "Android":
+            self._add_definition("USE_NNAPI_BUILTIN", self.options.with_nnapi)
+
+        self._add_definition("USE_RKNPU", False)
+        self._add_definition("USE_DNNL", self.options.with_dnnl)
+        self._add_definition("USE_FEATURIZERS", False)
+        self._add_definition("DEV_MODE", False)
+        self._add_definition(self._cmake_rt_def, self.options.with_static_rt)
+
+        self._add_definition("USE_PREINSTALLED_EIGEN", True)
+        eigen_includedir = os.path.join(self.deps_cpp_info["eigen"]
+                                        .include_paths[0], "eigen3")
+        self._cmake.definitions["eigen_SOURCE_PATH"] = eigen_includedir
+
+        self._add_definition("USE_TVM", self.options.with_tvm)
+        self._add_definition("USE_LLVM", self.options.with_tvm_llvm)
+
+        self._add_definition("USE_AVX", False)
+        self._add_definition("USE_AVX2", False)
+        self._add_definition("USE_AVX512", False)
+
+        self._add_definition("USE_OPENMP", False)
```

Comment:
shouldn't `with_openmp` be used here or anywhere else?

Reply:
it should, but before I need to be able pass tests at least for basic setup...